src/components/Sidebar/Sidebar.tsx

```diff
@@ -56,16 +56,17 @@ const Sidebar = ({ patient, disabled, savedStudies, filterOptions }: SidebarProp
     travelDistance: (query.travelDistance as string) || '',
     age: (query.age as string) || '',
     gender: (query.gender as string) || '',
-    cancerType: ensureNamedSNOMEDCode(query.cancerType),
-    cancerSubtype: ensureNamedSNOMEDCode(query.cancerSubtype),
+    cancerType: ensureCodedValueType(query.cancerType as string),
+    cancerSubtype: ensureCodedValueType(query.cancerSubtype as string),
     metastasis: ensureArray(query.metastasis),
-    stage: (query.stage as string) || null,
+    //stage: (query.stage as string) || null,
+    stage: ensureCodedValueType(query.stage as string),
     ecogScore: (query.ecogScore as string) || null,
     karnofskyScore: (query.karnofskyScore as string) || null,
-    biomarkers: ensureArray(query.biomarkers),
-    radiation: ensureArray(query.radiation),
-    surgery: ensureArray(query.surgery),
-    medications: ensureArray(query.medications),
+    biomarkers: parseCodedValueArray(query.biomarkers as string),
+    radiation: parseCodedValueArray(query.radiation as string),
+    surgery: parseCodedValueArray(query.surgery as string),
+    medications: parseCodedValueArray(query.medications as string),
```

Comment:
I'm not sure why `as string` is required but that's an immediately code smell. The functions should be updated to use whatever the actual type is, and if it's a string, then the type of the object should be fixed.

Reply:
I've pushed a change that resolves this by allowing the parse methods to accept arrays.